src/main/java/com/startstepszalando/ecommerceshop/user/controller/UserController.java

```diff
@@ -2,50 +2,101 @@
+import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.Date;
+
 @RestController
 @RequestMapping("/api/users")
 @RequiredArgsConstructor
 public class UserController {
 
     private final UserService userService;
+    private final AuthenticationManager authenticationManager;
+    private final JwtService jwtService;
 
     @Operation(summary = "Register a new user", responses = {
             @ApiResponse(responseCode = "200", description = "User registered successfully",
                     content = @Content(mediaType = "application/json",
                             schema = @Schema(implementation = AuthenticationResponse.class))),
-            @ApiResponse(responseCode = "400", description = "Invalid user details or validation error",
-                    content = @Content(mediaType = "application/json",
-                            schema = @Schema(implementation = RegistrationException.class))),
-            @ApiResponse(responseCode = "401", description = "Invalid credentials or token",
-                    content = @Content(mediaType = "application/json",
-                            schema = @Schema(implementation = InvalidCredentialsException.class))),
             @ApiResponse(responseCode = "404", description = "User not found",
                     content = @Content(mediaType = "application/json",
                             schema = @Schema(implementation = UserNotFoundException.class))),
-            @ApiResponse(responseCode = "409", description = "Duplicate user detected",
+            @ApiResponse(responseCode = "400", description = "Duplicate user detected",
                     content = @Content(mediaType = "application/json",
                             schema = @Schema(implementation = DuplicateUserException.class))),
             @ApiResponse(responseCode = "500", description = "Internal server error or unexpected error during registration",
                     content = @Content(mediaType = "application/json",
                             schema = @Schema(implementation = Exception.class)))
     })
+//    @PostMapping("/register")
+//    public ResponseEntity<?> registerUser(@Valid @RequestBody UserRegistrationRequest request) throws DuplicateUserException {
+//        AuthenticationResponse response = userService.registerUser(request);
+//        return ResponseEntity.ok(response);
+//    }
+
```

Comment:
do we still need this code?

Reply:
Forgot to remove it 😅 . I'll remove it and then merge.